label_studio/io_storages/migrations/0017_auto_20240731_1638.py

```diff
@@ -1,40 +1,99 @@
+        "column_name": "annotation_id"
+    },
+    {
+        "model_name": "localfilesexportstoragelink",
+        "table_name": "io_storages_localfilesexportstoragelink",
+        "index_name": "io_storages_localfilesexportstoragelink_annotation_id_fc4f9825",
+        "fk_constraint": "io_storages_localfil_annotation_id_fc4f9825_fk_task_comp",
+        "column_name": "annotation_id"
+    },
+    {
+        "model_name": "redisexportstoragelink",
+        "table_name": "io_storages_redisexportstoragelink",
+        "index_name": "io_storages_redisexportstoragelink_annotation_id_8547e508",
+        "fk_constraint": "io_storages_redisexp_annotation_id_8547e508_fk_task_comp",
+        "column_name": "annotation_id"
+    },
+    {
+        "model_name": "s3exportstoragelink",
+        "table_name": "io_storages_s3exportstoragelink",
+        "index_name": "io_storages_s3exportstoragelink_annotation_id_729994fe",
+        "fk_constraint": "io_storages_s3export_annotation_id_729994fe_fk_task_comp",
+        "column_name": "annotation_id"
+    }
+]
 
 
 class Migration(migrations.Migration):
+    atomic = False
 
     dependencies = [
         ('tasks', '0047_merge_20240318_2210'),
         ('io_storages', '0016_add_aws_sse_kms_key'),
     ]
 
-    operations = [
-        migrations.AlterField(
-            model_name='azureblobexportstoragelink',
-            name='annotation',
-            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='io_storages_azureblobexportstoragelink', to='tasks.annotation'),
-        ),
-        migrations.AlterField(
-            model_name='gcsexportstoragelink',
-            name='annotation',
-            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='io_storages_gcsexportstoragelink', to='tasks.annotation'),
-        ),
-        migrations.AlterField(
-            model_name='localfilesexportstoragelink',
-            name='annotation',
-            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='io_storages_localfilesexportstoragelink', to='tasks.annotation'),
-        ),
-        migrations.AlterField(
-            model_name='redisexportstoragelink',
-            name='annotation',
-            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='io_storages_redisexportstoragelink', to='tasks.annotation'),
-        ),
-        migrations.AlterField(
-            model_name='s3exportstoragelink',
-            name='annotation',
-            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='io_storages_s3exportstoragelink', to='tasks.annotation'),
-        ),
-    ]
+    operations = []
+
+    for table in tables:
+        index_sql = create_index_sql(table['table_name'], table['index_name'], table['column_name'])
+        fk_sql = create_fk_sql(table['table_name'], table['fk_constraint'], table['column_name'], "task_completion", "id")
```

Comment:
```
        index_sql = create_index_sql(table['table_name'], table['index_name'], table['column_name'])
        fk_sql = create_fk_sql(table['table_name'], table['fk_constraint'], table['column_name'], "task_completion", "id")
```
This should be moved to after the IS_SQLITE check, as the usage is only when it's not Sqlite.